File: src/main/java/hiiragi283/material/HiiragiProxy.java
package hiiragi283.material;

import net.minecraftforge.fml.common.event.*;

public interface HiiragiProxy {

    void onConstruct(FMLConstructionEvent event);

    void onPreInit(FMLPreInitializationEvent event);

    void onInit(FMLInitializationEvent event);

    void onPostInit(FMLPostInitializationEvent event);

    void onComplete(FMLLoadCompleteEvent event);

}

File: src/main/kotlin/hiiragi283/material/HiiragiProxy.kt
package hiiragi283.material

import net.minecraftforge.fml.common.event.*

interface HiiragiProxy {

    fun onConstruct(event: FMLConstructionEvent) {}

    fun onPreInit(event: FMLPreInitializationEvent) {}

    fun onInit(event: FMLInitializationEvent) {}

    fun onPostInit(event: FMLPostInitializationEvent) {}

    fun onComplete(event: FMLLoadCompleteEvent) {}

}